engine/grid.py

```diff
@@ -303,70 +304,106 @@ def find_border_nodes(self):
         cols = self.nodes.shape[1]
         rows = self.nodes.shape[0]
         leftmost_node = None
+        rightmost_node = None
         leftmost_node_pos = None
+        rightmost_node_pos = None
+        
         for row in range(rows):
             for col in range(cols):
                 node = self.nodes[row][col]
+                #border node 1 left active node 
                 if node.is_active and self.is_on_border(row, col, rows-1, cols-1):
```

Comment:
let's add documentation / a comment explaining how this if statement is different from the node below it, i.e "Checking if current node is active and is on the left side of the border" or something

Reply:
I forgot what this comment is, but I think the if statement is explained in the comment below? I can delete this comment since it is unused

Reply:
The if statement is documented in a comment below, I have no idea what this comment is about. I will go delete it